api/envoy/config/bootstrap/v3/bootstrap.proto

```diff
@@ -165,15 +165,20 @@ message Bootstrap {
   // Configuration for internal processing of stats.
   metrics.v3.StatsConfig stats_config = 13;
 
-  // Optional duration between flushes to configured stats sinks. For
-  // performance reasons Envoy latches counters and only flushes counters and
-  // gauges at a periodic interval. If not specified the default is 5000ms (5
-  // seconds).
-  // Duration must be at least 1ms and at most 5 min.
-  google.protobuf.Duration stats_flush_interval = 7 [(validate.rules).duration = {
-    lt {seconds: 300}
-    gte {nanos: 1000000}
-  }];
+  oneof flush_mode {
```

Comment:
https://github.com/envoyproxy/envoy/blob/master/api/API_VERSIONING.md#backwards-compatibility disallows adding new oneofs in a major version. Instead use the `oneof_promotion` API annotation.